TcpIpInput.h

```diff
@@ -1,18 +1,28 @@
 #ifndef TCPIPINPUT_H
 #define TCPIPINPUT_H
 #include "Input.h"
+#include <unistd.h>
+#include <stdio.h>
+#include <sys/socket.h>
+#include <stdlib.h>
+#include <netinet/in.h>
+#include <string.h>
 
 
 
 
 class TcpIpInput : public Input {
 private:
     short mPort;
+    struct sockaddr_in mAddress;
+    char mBuffer[1024] = {0};
+    int mServerFile;
 
 public:
-    TcpIpInput(short port);
+    TcpIpInput(short port, unsigned short domain, unsigned int type);
```

Comment:
Could you add a short explanation for `domain` and `type` arguments in a comment above the declaration? Also, I guess that in 99% the same values for `domain` and `type` would be used, so could you add some defaults?

Reply:
Another problem is that `port` is `short`, which limits the port number to 32767, but TCP ports can reach 65535. This is why `htons` takes uint16_t. Can you fix the data type?

Reply:
done